proposals/NNNN-unlock-existentials-with-fixed-associated-types.md

```diff
@@ -0,0 +1,237 @@
+# Unlock Existentials with Fixed Associated Types
+
+* Proposal: [SE-NNNN](NNNN-unlock-existentials-with-fixed-associated-types.md)
+* Authors: [Filip Sakel](https://github.com/filip-sakel), [Anthony Latsis](https://github.com/AnthonyLatsis)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+## Introduction
+
+Today, Swift protocols are divided into two categories: those that _can_ be used as fully-fledged types, and those that can only be used as generic constraints because they have `Self` or associated type requirements. However, in some cases where a protocol inherits and fixes the associated types of another protocol - that doesn't have `Self` requirements, this constraint seems rather unintuitive. This proposal aims to relax this needless contraint allowing such protocols to be used as Types.
+
+## Motivation
+
+Currently, any protocol that has associated type requirements is not allowed to be used as a Type. That's a sensible constraint; however, considering that the associated types of some protocols are known, this restriction can become quite frustrating: 
+
+```swift
+protocol User: Identifiable 
+    where ID == String {
+    
+    var username: String { get }
+    var displayName: String { get }
+}
+
+extension User {
+    var id: String {
+        username
+    }
+}
+```
+
+Many would rightfully assume that `User` could be used as a Type, since we specify that the `ID` asscoated type inherited from [`Identifiable`](https://github.com/apple/swift-evolution/blob/master/proposals/0261-identifiable.md) is `String`. 
```

Comment:
>```diff
>+ since we specify that the `ID` associated type inherited from `Identifiable` is `String`.
>```

For the sake of being more accurate, WDYT of rephasing this to
```diff
+ since the only associated type `ID` is known to be equal to `String` via a
+ same-type constraint on the `User` protocol.
```